src/App.js

```diff
@@ -1,25 +1,73 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from 'react';
+import styles from './styles/App.module.css';
+import { spotifySearch } from './Constants/httpRequests.js';
+import { handleReauthorization } from './Constants/authorization.js';
+import Header from './components/Header.jsx';
+import LoginToSpotify from './components/LoginToSpotify.jsx';
+import SearchBar from './components/SearchBar.jsx';
+import SearchResults from './components/SearchResults.jsx';
+import Playlist from './components/Playlist.jsx';
+
+export default function App() {
+  const [searchData, setSearchData] = useState([]);
+  const [playlistTracks, setPlaylistTracks] = useState([]);
+  const [trackUris, setTrackUris] = useState([]);
+
+  const handleSpotifySearch = async (searchInput) => {
+    const searchData = await spotifySearch(searchInput);
+    setSearchData(searchData);
+  };
+
+  // Add track to playlist
+  const addSearchResultToPlaylist = (track) => {
+    // Check for token expiration before adding track to playlist
+    handleReauthorization();
+    
+    if (!playlistTracks.some((playlistTrack) => playlistTrack.id === track.id)) {
+      setPlaylistTracks((prevTracks) => [...prevTracks, track]);
+
+      // Update trackUris with the new track's URI
+      setTrackUris((prevTrackUris) => [...prevTrackUris, track.uri]);
+    }
+  };
+
+  // Remove track from playlist
+  const removeSearchResultFromPlaylist = (trackId) => {
+    // Check for token expiration before removing track from playlist
+    handleReauthorization();
+  
+    setPlaylistTracks((prevTracks) => {
+    // Filter out the removed track from the playlist
+    const updatedTracks = prevTracks.filter((track) => track.id !== trackId.id);
+
+    // Update trackUris by filtering out the URI of the removed track
+    setTrackUris((prevTrackUris) => prevTrackUris.filter((uri) => uri !== trackId.uri));
```

Comment:
It seems odd to have setTrackUris called inside of setPlaylistTracks. Can you move it outside of the setPlaylistTracks function?